### PR TITLE
Documentation for alitv.pl

### LINKED ISSUE #41: Main script needs documentation

### PR DESCRIPTION
**Pull request by greatfireball**
Adds the documentation for `alitv.pl` including the description of the YML file.

Fixes #41

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/AliTVTeam/AliTV-perl-interface/pull/128" target="_blank">ReviewNinja

File: README.md [hunk 1]
```diff
@@ -1,3 +1,188 @@
...
+- --overwrite or --force   Overwrite existing project.yml or output.json files
+
+    Default behaviour is to keep existing project yml and json files. If
+    `--overwrite` or `--force` is specified, the files will be
+    overwritten. Overwritting can be expicitly disabled by
+    `--no-overwrite` or `--no-force` parameter.
+
+# PARAMETERS
+
+The parameters might be a single YML file or at least to sequence
```
Comment (iimog) [inline README.md]: typo: to -> two

File: README.md [hunk 2]
```diff
@@ -1,3 +1,188 @@
...
+    alitv.pl [OPTIONS] options.yml
+
+    # OR
+
+    # easy alternative including the generation of a yml file
+    alitv.pl [OPTIONS] *.fasta
+
+# DESCRIPTION
+
+The script creates the required JSON file to run AliTV. The output
```
Comment (iimog) [inline README.md]: run -> visualize with/import into ?

Comment (iimog) [reply README.md]: or just: "to run" -> for

File: README.md [hunk 3]
```diff
@@ -1,3 +1,188 @@
...
+- --logfile   Log file
+
+    The name of the log file. If non is provided, the log file name will
+    be based on the project name.
+
+- --overwrite or --force   Overwrite existing project.yml or output.json files
+
+    Default behaviour is to keep existing project yml and json files. If
+    `--overwrite` or `--force` is specified, the files will be
+    overwritten. Overwritting can be expicitly disabled by
```
Comment (iimog) [inline README.md]: typo: Overwritting -> Overwriting

File: README.md [hunk 4]
```diff
@@ -1,3 +1,188 @@
...
+
+# Description of the YML file format
+
+The YML file contains the whole project description. Currently, it
+supports three different sections:
+
+## Section genomes
+
+Within the genomes section, each genome has to be declared. One genome
+need to have a unique name, at least one sequence file, and
```
Comment (iimog) [inline README.md]: typo: need -> needs

File: README.md [hunk 5]
```diff
@@ -1,3 +1,188 @@
...
+the file will be given for further runs.
+
+# Description of the YML file format
+
+The YML file contains the whole project description. Currently, it
+supports three different sections:
+
+## Section genomes
+
+Within the genomes section, each genome has to be declared. One genome
```
Comment (iimog) [inline README.md]: One -> Each ?

File: README.md [hunk 6]
```diff
@@ -1,3 +1,188 @@
...
+need to have a unique name, at least one sequence file, and
+additionally can have feature definitions.
+
+- `name` key defines the name of the genome and has to be unique.
+- `sequence_files` key contains a list of sequence files which define
+the sequence(s) of the genome
+- `feature_files` keys contains different features. Each is defined as
```
Comment (iimog) [inline README.md]: typo: keys -> key

File: README.md [hunk 7]
```diff
@@ -1,3 +1,188 @@
...
+files. In case a YML file is specified, the project name will be set
+accordingly to the basename of the YML file (without path information
+and without the .yml suffix). In case of multiple sequence files, a
+YML file will be created based on the default template. The name of
+the file will be given for further runs.
+
+# Description of the YML file format
+
+The YML file contains the whole project description. Currently, it
+supports three different sections:
```
Comment (iimog) [inline README.md]: I count four sections: genomes, tree, features, alignment

File: README.md [hunk 8]
```diff
@@ -1,3 +1,188 @@
...
+            visible: 1
+        ndh:
+            color: "#ff006e"
+            form: rect
+            height: 30
+            visible: 1
+
+## Section alignment
+
+This section contains of the specification of the alignment program
```
Comment (iimog) [inline README.md]: typo: "contains of" -> "contains"

File: bin/alitv.pl [hunk 9]
```diff
@@ -247,6 +247,151 @@ =head1 PARAMETERS
 YML file will be created based on the default template. The name of
 the file will be given for further runs.
 
+=head1 Description of the YML file format
+
+The YML file contains the whole project description. Currently, it
+supports three different sections:
```
Comment (iimog) [inline bin/alitv.pl]: See above: three -> four

Comment (iimog) [reply bin/alitv.pl]: Generally most of the comments from above apply to this file as well as it contains identical text in some places

**Review by iimog — CHANGES_REQUESTED** [review]
Nice! Very good work, thanks a lot. The documentation is extensive and easy to understand due to good examples. Please fix the typos as indicated.

File: README.md (hunk 2 above)
Comment (greatfireball) [reply README.md]: check

File: README.md (hunk 3 above)
Comment (greatfireball) [reply README.md]: done

File: README.md (hunk 1 above)
Comment (greatfireball) [reply README.md]: check

File: README.md (hunk 7 above)
Comment (greatfireball) [reply README.md]: :) check

File: README.md (hunk 5 above)
Comment (greatfireball) [reply README.md]: check

File: README.md (hunk 6 above)
Comment (greatfireball) [reply README.md]: check

File: README.md (hunk 8 above)
Comment (greatfireball) [reply README.md]: check

File: bin/alitv.pl (hunk 9 above)
Comment (greatfireball) [reply bin/alitv.pl]: derived from changed perldoc

**Review by iimog — APPROVED** [review]

**Pull request merged**